src/DataProtection/Azure.Storage.Blob/src/Microsoft.AspNetCore.DataProtection.Azure.Storage.Blob.csproj

```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Description>Microsoft Azure Blob storage support as key store.</Description>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PackageTags>aspnetcore;dataprotection;azure;blob</PackageTags>
+    <IsShippingPackage>true</IsShippingPackage>
+    <UseLatestPackageReferences>true</UseLatestPackageReferences>
+    <PreReleaseVersionLabel>$(AzureSDKPreReleaseVersionLabel)</PreReleaseVersionLabel>
```

Comment:
I don't think this is working. Looking at the public CI artifacts (which perhaps are wrong) I see: `3.1.1-ci`. That indicates two things to me:

1. The `.1` patch version is applied. We want these to be `.0`.
1. The `-preview1` label is being lost, though again, perhaps that's a problem in the public CI.

@dotnet/aspnet-build do you have context on this? The same problem likely applies to the other two PRs (https://github.com/dotnet/aspnetcore/pull/17041, https://github.com/dotnet/extensions/pull/2618)

Reply:
The `-ci` is expected I believe, and `-preview1` should be applied in official builds. If you look at the other already shipped packages, you'll see they don't have `-ci`.

Reply:
The setup looks correct w.r.t. `$(PreReleaseVersionLabel)` and `$(DotNetFinalVersionKind)`.

> We want these to be `3.1.0`

Need to change the `$(VersionPrefix)` for that. I suggest the following in this project
```xml
<VersionPrefix>$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion).0</VersionPrefix>
```

Reply:
Sounds good. To be clear, this weird versioning will only hold true until we release the `.0`. Once that's done, we can just clean all the up and version them ahead to `3.1.x` (whatever the next `x` is). I'll make a note to track that for follow up after these packages go out.

Reply:
@dougbu Do you know how to do the same thing for Extensions? Setting VersionPrefix doesn't seem to affect the nupkg version

Reply:
I believe there's an Arcade or .NET Core SDK setting that causes it to override `$(VerisonPrefix)`. IIRC that's set in dotnet/aspnetcore but not dotnet/extensions. Might be `$(IncludeSourceRevisionInInformationalVersion)`.

I didn't check a 3.x Arcade SDK but their logic in 'master' for calculating `$(VersionPrefix)` is in tools\Version.BeforeCommonTargets.targets and seems to no-op when the property is already set and `$(PrereleaseVersionLabel)` is non-empty.

Bottom line, no, I'm not sure and suggest you complete the research I started 😈